ehrql/query_model/nodes.py

```diff
@@ -238,6 +238,9 @@ class Exists(OneRowPerPatientSeries[bool]):
     class Count(OneRowPerPatientSeries[int]):
         source: Frame
 
+    class CountDistinct(AggregatedSeries[int]):
+        source: Series[Comparable]
```

Comment:
This can actually be `Series[Any]`. The `Comparable` class (which I don't think is a great name but it fits with conventions used elsewhere) is for types which have `>` and `<` defined. But here we only need types which can be compared as equal or not, and all types support that.

Reply:
thanks!